x-pack/plugins/discover_log_explorer/public/components/dataset_selector/dataset_selector.tsx

```diff
@@ -0,0 +1,176 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import { EuiContextMenu, EuiHorizontalRule } from '@elastic/eui';
+import { dynamic } from '../../../common/dynamic';
+import { useIntersectionRef } from '../../hooks/use_intersection_ref';
+import {
+  contextMenuStyles,
+  DATA_VIEW_POPOVER_CONTENT_WIDTH,
+  integrationsLabel,
+  INTEGRATION_PANEL_ID,
+  uncategorizedLabel,
+  UNMANAGED_STREAMS_PANEL_ID,
+} from './constants';
+import { useDatasetSelector } from './state_machine/use_dataset_selector';
+import { DatasetsPopover } from './sub_components/datasets_popover';
+import { DatasetSkeleton } from './sub_components/datasets_skeleton';
+import { SearchControls } from './sub_components/search_controls';
+import { DatasetSelectorProps } from './types';
+import { buildIntegrationsTree, setIntegrationListSpy } from './utils';
+
+/**
+ * Lazy load hidden components
+ */
+const DatasetsList = dynamic(() => import('./sub_components/datasets_list'), {
+  fallback: <DatasetSkeleton />,
+});
+const IntegrationsListStatus = dynamic(() => import('./sub_components/integrations_list_status'));
+
+export function DatasetSelector({
+  datasets,
+  datasetsError,
+  integrations,
+  integrationsError,
+  isLoadingIntegrations,
+  isLoadingStreams,
+  onIntegrationsLoadMore,
+  onIntegrationsReload,
+  onIntegrationsSearch,
+  onIntegrationsSort,
+  onIntegrationsStreamsSearch,
+  onIntegrationsStreamsSort,
+  onStreamSelected,
+  onStreamsEntryClick,
+  onUnmanagedStreamsReload,
+  onUnmanagedStreamsSearch,
+  onUnmanagedStreamsSort,
+  title,
+}: DatasetSelectorProps) {
+  const {
+    isOpen,
+    panelId,
+    search,
+    changePanel,
+    scrollToIntegrationsBottom,
+    searchByName,
+    selectDataset,
+    sortByOrder,
+    togglePopover,
+  } = useDatasetSelector({
+    onIntegrationsLoadMore,
+    onIntegrationsReload,
+    onIntegrationsSearch,
+    onIntegrationsSort,
+    onIntegrationsStreamsSearch,
+    onIntegrationsStreamsSort,
+    onUnmanagedStreamsSearch,
+    onUnmanagedStreamsSort,
+    onUnmanagedStreamsReload,
+    onStreamSelected,
+  });
+
+  const [setSpyRef] = useIntersectionRef({ onIntersecting: scrollToIntegrationsBottom });
+
+  const { items: integrationItems, panels: integrationPanels } = useMemo(() => {
+    const datasetsItem = {
+      name: uncategorizedLabel,
+      onClick: onStreamsEntryClick,
+      panel: UNMANAGED_STREAMS_PANEL_ID,
+    };
+
+    const createIntegrationStatusItem = () => ({
+      disabled: true,
+      name: (
+        <IntegrationsListStatus
+          error={integrationsError}
+          integrations={integrations}
+          onRetry={onIntegrationsReload}
+        />
+      ),
+    });
+
+    if (!integrations) {
+      return {
+        items: [datasetsItem, createIntegrationStatusItem()],
```

Comment:
Would it make sense to move the "uncategorized" item to the end of the list? The hope is that it would be the exception that a user would have to make use of it.

Reply:
That's an option, my reason for adding that on top was that they won't be able to know about that if they don't scroll till the end of the list.
Also, since the list uses infinite loading, would be a bit weird maybe to add new integrations just before that last entry?
The best scenario would be to maybe have it sticky, but I'm not sure if the `EuiContextMenu` support that and would work the same way.

Reply:
Right, I forgot about the infinite loading. Let's ask others for feedback on that. (@formgeist or @isaclfreire maybe? :innocent:)

Reply:
I agree it's a little bit of an eyesore to see Uncategorized as the first item in the menu. I've seen in the EUI component docs that it may support a divider, but not sure about sticking it to the bottom. Not a big deal for the first iteration.